velox/functions/prestosql/tests/ArrayCumSumTest.cpp

```diff
@@ -0,0 +1,183 @@
+      {{},
+       {1, 3, 6},
+       {-9, std::nullopt, std::nullopt},
+       {std::nullopt, std::nullopt}});
+  testArrayCumSum<float>(expected, input);
+}
+
+TEST_F(ArrayCumSumTest, double) {
+  auto input = makeNullableArrayVector<double>(
+      {{},
+       {1, 2, 3},
+       {-9, std::nullopt, std::numeric_limits<double>::max()},
+       {std::nullopt, 3}});
+  auto expected = makeNullableArrayVector<double>(
+      {{},
+       {1, 3, 6},
+       {-9, std::nullopt, std::nullopt},
+       {std::nullopt, std::nullopt}});
+  testArrayCumSum<double>(expected, input);
+}
+
+TEST_F(ArrayCumSumTest, bigintOverflow) {
+  constexpr int64_t kMin = std::numeric_limits<int64_t>::min();
+  constexpr int64_t kMax = std::numeric_limits<int64_t>::max();
+
+  // Overflow on 1 + kMax.
+  auto input = makeNullableArrayVector<int64_t>({{1, kMax, 2, 3}});
+  VELOX_ASSERT_THROW(
+      evaluate("array_cum_sum(c0)", makeRowVector({input})),
+      "integer overflow: 1 + 9223372036854775807");
+
+  // Overflow on -1 + kMin.
+  input = makeNullableArrayVector<int64_t>({{-1, kMin, -2, -3}});
+  VELOX_ASSERT_THROW(
+      evaluate("array_cum_sum(c0)", makeRowVector({input})),
+      "integer overflow: -1 + -9223372036854775808");
+
+  // Overflow for array containing consecutive kMin values.
+  input = makeNullableArrayVector<int64_t>({{kMax, kMin, kMin}});
+  VELOX_ASSERT_THROW(
+      evaluate("array_cum_sum(c0)", makeRowVector({input})),
+      "integer overflow: -1 + -9223372036854775808");
+
+  // No overflow when function is called on alternating kMax, kMin values.
+  input = makeNullableArrayVector<int64_t>(
+      {{kMax, kMin, kMax, kMin},
+       {kMax, kMin},
+       {kMax, kMin, kMax, kMin, kMax, kMin}});
+  auto expected = makeNullableArrayVector<int64_t>(
+      {{kMax, -1, kMax - 1, -2},
+       {kMax, -1},
+       {kMax, -1, kMax - 1, -2, kMax - 2, -3}});
+  testArrayCumSum<int64_t>(expected, input);
+}
+
+TEST_F(ArrayCumSumTest, doubleLimits) {
+  constexpr double kInf = std::numeric_limits<double>::infinity();
+  constexpr double kNan = std::numeric_limits<double>::quiet_NaN();
+  constexpr double kLowest = std::numeric_limits<double>::lowest();
+  constexpr double kMax = std::numeric_limits<double>::max();
+
+  auto input = makeNullableArrayVector<double>(
+      {{1, kInf, 123.456},
+       {1, kNan, 123.456},
+       {1, kLowest, 123.456},
+       {1, kMax, 123.456},
+       {kLowest, kMax, 123.456},
+       {kMax, kLowest, 123.456},
+       {1, kMax, kInf},
+       {1, kLowest, kInf},
+       {kLowest, kNan, 5},
+       {kMax, kNan, 5}});
+  auto expected = makeNullableArrayVector<double>(
+      {{1, kInf, kInf},
+       {1, kNan, kNan},
+       {1, kLowest, kLowest},
+       {1, kMax, kMax},
+       {kLowest, 0, 123.456},
+       {kMax, 0, 123.456},
+       {1, kMax, kInf},
+       {1, kLowest, kInf},
+       {kLowest, kNan, kNan},
+       {kMax, kNan, kNan}});
+  testArrayCumSum<double>(expected, input);
+}
+} // namespace
```

Comment:
The clang-format we run in OS is missing a bunch of rules, in part because we're running an old version until we upgrade to CentOS 9.

Could you add a newline at the end of this file?  (Once we upgrade I'll make "make format-fix" do it for you)

Reply:
Updated, thanks @kevinwilfong.

Reply:
That's expected.  Thanks, it looks like passing the linting now, I'll land it once the tests pass.